### ISSUE TITLE
Issues using with Rider EAP

### ISSUE THREAD

**Issue by danwalmsley** [ForNeVeR/AvaloniaRider#98]
Iv been trying to test with the new Rider EAP builds which have better support for Avalonia...

I cant get it to work at all with .axaml extension. If I use .xaml I see a previewer area, but it never loads.

I will post some related exceptions which may or may not be normal.

**Comment by danwalmsley** [comment]
```
Catch

java.lang.IllegalArgumentException: Value already exists: com.jetbrains.rdclient.daemon.FrontendMarkupContributor@523ef585
	at com.jetbrains.rd.util.CollectionExKt$addUnique$2.invoke(CollectionEx.kt:16)
	at com.jetbrains.rd.util.CollectionExKt$addUnique$2.invoke(CollectionEx.kt)
	at com.jetbrains.rd.util.lifetime.LifetimeDefinition.executeIfAlive(RLifetime.kt:399)
	at com.jetbrains.rd.util.CollectionExKt.addUnique(CollectionEx.kt:15)
	at com.jetbrains.rdclient.daemon.components.FrontendMarkupHost$Companion.a(FrontendMarkupHost.kt:32)
	at com.jetbrains.rdclient.daemon.components.FrontendMarkupHost$Companion.access$putMarkupContributor(FrontendMarkupHost.kt:21)
	at com.jetbrains.rdclient.daemon.components.FrontendMarkupHost.a(FrontendMarkupHost.kt:81)
	at com.jetbrains.rdclient.daemon.components.FrontendMarkupHost.access$createProtocolDocumentMarkupModel(FrontendMarkupHost.kt:20)
	at com.jetbrains.rdclient.daemon.components.FrontendMarkupHost$1.invoke(FrontendMarkupHost.kt:48)
	at com.jetbrains.rdclient.daemon.components.FrontendMarkupHost$1.invoke(FrontendMarkupHost.kt:20)
	at com.jetbrains.rd.util.reactive.IViewableMap$view$2.invoke(ViewableCollections.kt:146)
	at com.jetbrains.rd.util.reactive.IViewableMap$view$2.invoke(ViewableCollections.kt:64)
	at com.jetbrains.rd.util.reactive.IViewableMap$view$1.invoke(ViewableCollections.kt:108)
	at com.jetbrains.rd.util.reactive.IViewableMap$view$1.invoke(ViewableCollections.kt:64)
	at com.jetbrains.rd.util.reactive.IViewableMap$adviseAddRemove$1.invoke(ViewableCollections.kt:126)
	at com.jetbrains.rd.util.reactive.IViewableMap$adviseAddRemove$1.invoke(ViewableCollections.kt:64)
	at com.jetbrains.rd.util.reactive.Signal.fire(Signal.kt:32)
	at com.jetbrains.rd.util.reactive.ViewableMap.put(ViewableMap.kt:29)
	at com.jetbrains.rdclient.document.FrontendDocumentHost.viewDocumentModel(FrontendDocumentHost.kt:89)
	at com.jetbrains.rdclient.document.FrontendDocumentHost$1$$special$$inlined$runReadAction$1.compute(actions.kt:67)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:871)
	at com.jetbrains.rdclient.document.FrontendDocumentHost$1.invoke(FrontendDocumentHost.kt:215)
	at com.jetbrains.rdclient.document.FrontendDocumentHost$1.invoke(FrontendDocumentHost.kt:30)
	at com.jetbrains.rd.util.reactive.IViewableMap$view$2.invoke(ViewableCollections.kt:146)
	at com.jetbrains.rd.util.reactive.IViewableMap$view$2.invoke(ViewableCollections.kt:64)
	at com.jetbrains.rd.util.reactive.IViewableMap$view$1.invoke(ViewableCollections.kt:108)
	at com.jetbrains.rd.util.reactive.IViewableMap$view$1.invoke(ViewableCollections.kt:64)
	at com.jetbrains.rd.util.reactive.IViewableMap$adviseAddRemove$1.invoke(ViewableCollections.kt:126)
	at com.jetbrains.rd.util.reactive.IViewableMap$adviseAddRemove$1.invoke(ViewableCollections.kt:64)
	at com.jetbrains.rd.util.reactive.Signal.fire(Signal.kt:32)
	at com.jetbrains.rd.util.reactive.ViewableMap.put(ViewableMap.kt:29)
	at com.jetbrains.rd.framework.impl.RdMap$put$1.invoke(RdMap.kt:175)
	at com.jetbrains.rd.framework.base.RdReactiveBase.localChange$rd_framework(RdReactiveBase.kt:52)
	at com.jetbrains.rd.framework.impl.RdMap.put(RdMap.kt:175)
	at com.jetbrains.rdclient.document.FrontendDocumentHost.registerModelDocument(FrontendDocumentHost.kt:231)
	at com.jetbrains.rider.document.RiderDocumentHost.registerModelDocument(RiderDocumentHost.kt:64)
	at com.jetbrains.rdclient.editors.FrontendTextControlHost.bindEditor(FrontendTextControlHost.kt:393)
	at com.jetbrains.rdclient.editors.FrontendTextControlHost.editorCreated(FrontendTextControlHost.kt:159)
	at com.jetbrains.rdclient.editors.FrontendTextControlHost$MyEditorFactoryListener.editorCreated(FrontendTextControlHost.kt:143)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.lambda$createEditor$0(EditorFactoryImpl.java:203)
	at com.intellij.openapi.extensions.impl.ExtensionProcessingHelper.forEachExtensionSafe(ExtensionProcessingHelper.java:33)
	at com.intellij.openapi.extensions.ExtensionPointName.forEachExtensionSafe(ExtensionPointName.java:49)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:203)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.createEditor(EditorFactoryImpl.java:157)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorComponent.createEditor(TextEditorComponent.java:133)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorComponent.<init>(TextEditorComponent.java:79)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorImpl$PsiAwareTextEditorComponent.<init>(PsiAwareTextEditorImpl.java:107)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorImpl.createEditorComponent(PsiAwareTextEditorImpl.java:85)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorImpl.<init>(TextEditorImpl.java:52)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorImpl.<init>(PsiAwareTextEditorImpl.java:36)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorProvider.createEditor(PsiAwareTextEditorProvider.java:31)
	at me.fornever.avaloniarider.idea.editor.AvaloniaPreviewEditorProvider.createEditor(AvaloniaPreviewEditorProvider.kt:22)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.dumbModeFinished(FileEditorManagerImpl.java:219)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$2.lambda$exitDumbMode$0(FileEditorManagerImpl.java:165)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:216)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:24)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:199)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:324)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:85)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:134)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:47)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:190)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:974)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:841)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:451)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:822)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:499)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by danwalmsley** [comment]
```
java.lang.NoSuchMethodError: 'com.jetbrains.rider.runtime.DotNetRuntime com.jetbrains.rider.runtime.DotNetRuntime$Companion.detectRuntimeForProjectOrThrow(com.jetbrains.rider.model.RunnableProjectKind, com.jetbrains.rider.runtime.RiderDotNetActiveRuntimeHost, com.jetbrains.rider.debugger.DebuggerHelperHost, boolean, java.lang.String, java.lang.String)'
	at me.fornever.avaloniarider.previewer.MsBuildParameterCollector.getAvaloniaPreviewerParameters(MsBuildParameterCollector.kt:71)
	at me.fornever.avaloniarider.previewer.AvaloniaPreviewerSessionController.executePreviewerAsync(AvaloniaPreviewerSessionController.kt:201)
	at me.fornever.avaloniarider.previewer.AvaloniaPreviewerSessionController$executePreviewerAsync$1.invokeSuspend(AvaloniaPreviewerSessionController.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```

**Comment by ForNeVeR** [comment]
Hi!

First of all, the latest builds of the plugin are currently compatible with Rider EAP only (due to internal API changes, I wasn't able to support both Rider stable and EAP at the same time).

This plugin is not supposed to work with `.paml`/`.axaml`, yet. I was waiting for the relevant ReSharper / Rider changes (that have already landed, so I should make it work soon).

Regarding the issue itself: well, something problematic is happening; I'll take a look ASAP. Thanks for noticing!

(Hopefully, we'll get some integration tests before Rider 2020.3 release, so issues such as this one won't happen so often in future.)

**Comment by ForNeVeR** [comment]
@danwalmsley, I'm sorry, but I wasn't able to reproduce the issue. Which plugin version are you on?

**Comment by PieroCastillo** [comment]
@ForNeVeR  I think  I've the same problem, only shows a progress ring but never load the previewer, the plugin version is 1.0.0.363380077, and the Rider version is 2020.3 EAP 7

**Comment by ForNeVeR** [comment]
Ah, now I think I can see the issue. The mentioned method, `detectRuntimeForProjectOrThrow`, wasn't available in EAP 7 (well, it was, but with a different signature). I think you just need to update to EAP 8. (Unfortunately, it's not that easy to find old EAP builds for me to check it the plugin is actually broken there.)

Yes, during active Rider 2020.3 development, plugin was broken several times, which could happen. This is one of the reasons plugin isn't yet published in the release channel.

I expect the plugin to still be available for 2020.3 after its release, so the users won't have to follow EAP builds so closely.

If it won't be fixed for you in 2020.3 EAP 8, feel free to drop a comment; I'll reopen the issue and investigate further.

**Comment by SpyceTewan** [comment]
So .axaml isn't supported yet? .xaml previews fine for me, but my .axaml views don't ever preview. This means that i can only preview my styles right now
